pkg/resources/digitalocean/dns/dns.go

```diff
@@ -374,112 +372,70 @@ func (r *resourceRecordChangeset) ResourceRecordSets() dnsprovider.ResourceRecor
 		return nil, fmt.Errorf("failed to list domains: %v", err)
 	}
 
-	if err = handleResponse(resp); err != nil {
-		return nil, err
-	}
-
 	return domains, err
 }
 
 // createDomain creates a domain provided godo.DomainCreateRequest
 func createDomain(c *godo.Client, createRequest *godo.DomainCreateRequest) (*godo.Domain, error) {
-	domain, resp, err := c.Domains.Create(context.TODO(), createRequest)
+	domain, _, err := c.Domains.Create(context.TODO(), createRequest)
 	if err != nil {
 		return nil, err
 	}
 
-	if err = handleResponse(resp); err != nil {
-		return nil, err
-	}
-
 	return domain, nil
 }
 
 // deleteDomain deletes a domain given its name
 func deleteDomain(c *godo.Client, name string) error {
-	resp, err := c.Domains.Delete(context.TODO(), name)
+	_, err := c.Domains.Delete(context.TODO(), name)
 	if err != nil {
 		return err
 	}
 
-	if err = handleResponse(resp); err != nil {
-		return err
-	}
-
 	return nil
 }
 
 // getRecords returns a list of godo.DomainRecord given a zone name
 func getRecords(c *godo.Client, zoneName string) ([]godo.DomainRecord, error) {
-	records, resp, err := c.Domains.Records(context.TODO(), zoneName, &godo.ListOptions{})
+	records, _, err := c.Domains.Records(context.TODO(), zoneName, &godo.ListOptions{})
 	if err != nil {
 		return nil, err
 	}
 
-	if err = handleResponse(resp); err != nil {
-		return nil, err
-	}
-
 	return records, nil
 }
 
 // createRecord creates a record given an associated zone and a godo.DomainRecordEditRequest
 func createRecord(c *godo.Client, zoneName string, createRequest *godo.DomainRecordEditRequest) error {
-	_, resp, err := c.Domains.CreateRecord(context.TODO(), zoneName, createRequest)
+	_, _, err := c.Domains.CreateRecord(context.TODO(), zoneName, createRequest)
 	if err != nil {
 		return fmt.Errorf("error applying changeset: %v", err)
 	}
 
-	if err = handleResponse(resp); err != nil {
-		return err
-	}
-
 	return nil
 }
 
 // editRecord edits a record given an associated ozone and a godo.DomainRecordEditRequest
 func editRecord(c *godo.Client, zoneName string, recordID int, editRequest *godo.DomainRecordEditRequest) error {
-	_, resp, err := c.Domains.EditRecord(context.TODO(), zoneName, recordID, editRequest)
+	_, _, err := c.Domains.EditRecord(context.TODO(), zoneName, recordID, editRequest)
 	if err != nil {
 		return fmt.Errorf("error applying changeset: %v", err)
 	}
 
-	if err = handleResponse(resp); err != nil {
-		return err
-	}
-
 	return nil
 }
 
 // deleteRecord deletes a record given an associated zone and a record ID
 func deleteRecord(c *godo.Client, zoneName string, recordID int) error {
-	resp, err := c.Domains.DeleteRecord(context.TODO(), zoneName, recordID)
+	_, err := c.Domains.DeleteRecord(context.TODO(), zoneName, recordID)
 	if err != nil {
 		return fmt.Errorf("error applying changeset: %v", err)
 	}
 
-	if err = handleResponse(resp); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func handleResponse(resp *godo.Response) error {
```

Comment:
I take it that godo now handles this for you?

Reply:
Yes! That was actually always the case, just only recently found out :)